src/hook.js

```diff
@@ -19,22 +21,38 @@ function applyClassMap(value, classMap) {
   return classNames;
 }
 
-function rendersDOMComponent(component) {
-  return typeof component._currentElement.type === 'string';
+const { mountComponent, updateComponent } = ReactDOMComponent.prototype;
+
+function applyClassMapToElement(element, context) {
+  return cloneElement(element, {
+    className: applyClassMap(
+      element.props.className,
+      context[CLASSMAP_KEY]
+    ),
+  });
 }
 
-const { _processProps } = ReactCompositeComponent.Mixin;
-ReactCompositeComponent.Mixin._processProps = function(props) {
-  let processedProps = _processProps.call(this, props);
-  if (
-    rendersDOMComponent(this) &&
-    typeof processedProps.className !== 'undefined'
-  ) {
-    let context = this._context;
-    processedProps = {
-      ...processedProps,
-      className: applyClassMap(processedProps.className, context[CLASSMAP_KEY])
-    };
-  }
-  return processedProps;
+ReactDOMComponent.prototype.mountComponent = function(
```

Comment:
There might be a way around it by creating a react renderer, but it seems like a lot of trouble for not much gain.